enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,357 @@
+
+### API Extensions
+N/A
+
+### Risks and Mitigations
+
+* With this solution, the platform's housekeeping processes also runs on the shared CPUs.
+but the intent is to allocate those cpus to workload’s light-weight tasks,
+so having some latency is bearable.
+A way of mitigating that is checks whether it's possible to use workload partitioning
+to ensure the platform housekeeping processes don't run on the shared cpus.
+(This statement needs to be verified)
+
+* There is no risk that workload’s tasks run on dedicated OCP’s housekeeping reserved cpus,
+because only the shared cpus exposed via cgroups to the workload’s process.
+In other words, platform's housekeeping processes can run on shared cpus dedicated to workload’s light-weight tasks,
+but not the other way around.
+
+* mixed-cpus-node-plugin is NRI-based plugin.
+Bugs in the plugin or NRI framework might affect the containers at various lifecycle stages, e.g.,
+CreateContainer, UpdateContainer, etc.
+For mitigating such cases, it is possible to deactivate the node-plugin, and in severe cases it is possible to disable
+NRI completely at runtime-level.
+
+* From a performance perspective, the runtime communicates with the node-plugin via gRPC calls.
+Therefore, for every container creation, there's another gRPC call between the runtime and the node-plugin.
+We can't avoid this call even for containers that don't ask for shared CPUs, since runtime doesn't have any filtering
+mechanism, it sends all the containers through the NRI framework to node-plugin, and the filtering is done at the
+plugin side.
+
+### Drawbacks
+
+The way of how user should ask for shared CPUs in the pod spec is through device plugin.
+While this approach fits in the model of how the workload requires resources,
+better integration would require more invasive changes which are out of scope now.
+    
+## Design Details
+N/A
+
+### Test Plan
+
+#### Unit testing
+We will have unit-testing for the mixed-cpu-node-plugin to guarantee the basic functionality.
+
+#### E2E testing on mixed-cpu-node-plugin
+The behavior of the mixed-cpu-node-plugin will be validated in isolation (without NTO) by E2E tests
+
+#### E2E testing on NTO
+Validating a deployment process on NTO, verifying PerformanceProfile configuration and checking 
+the functionality of mixed-cpu-node-plugin (running the e2e tests from mixed-cpu-node-plugin) 
+
+### Graduation Criteria
+N/A
+
+#### Dev Preview -> Tech Preview
+N/A
+
+#### Tech Preview -> GA
+N/A
+
+#### Removing a deprecated feature
+* For two releases after shared-partition lands, a deprecation warning should appear when the annotation is used,
+asking the user to start using the `spec.cpu.shared` field for configuring the shared cpus.
+* Within the two releases, the annotation still takes precedence over the `spec.cpu.shared`.
+* After the two releases, the annotation would be ignored and `spec.cpu.shared` takes precedence.
```

Comment:
It strikes me that we are creating tech debt knowingly by implementing this downstream ahead of the upstream feature, why can't we wait to avoid this tech debt?

Reply:
the shared-partition is not an upstream feature.
See #1421